base/src/main/java/org/aya/concrete/visitor/StmtFolder.java

```diff
@@ -49,7 +49,7 @@
     return switch (stmt) {
       case Generalize g -> g.variables.foldLeft(acc, (a, v) -> foldVarDecl(a, v, v.sourcePos, noType()));
       case Command.Module m -> foldModuleDecl(acc, m.sourcePos(), new ModuleName.Qualified(m.name()));
-      case Command.Import i -> acc; // we are no need to fold `i.path()`
+      case Command.Import i -> foldModuleRef(acc, i.sourcePos(), i.path().asName());
```

Comment:
Do we need to `fold` the `ModuleName`s which `import` introduces? For example, `import Data::Nat as Nat` introduces a ModuleName `Nat`.

Reply:
Sure, but that requires some hacks in `Stmt.Import`, `Stmt.Open`. I decide to fix them in later patches.

Reply:
There are two bugs currently: 
- `Data::Option`, it used `import XXX as YYY` then `open YYY`, which results in a wrong cross-link for `YYY` (that's the problem in this comment).
- `SimpleModule`, it opens a local module.

I think the solution to them should be same: by recoding the really-qualified-path in `Stmt.Open`